pennylane/pauli/pauli_arithmetic.py

```diff
@@ -133,6 +134,8 @@ class PauliWord(dict):
     X(a) @ Y(2) @ Z(3)
     """
 
+    __array_priority__ = 1000
```

Comment:
I got this from [stackexchange](https://stackoverflow.com/a/44634634) and need this so the mul method of this class is prioritized when using autograd (not for jax, torch or tf), when removing it the autograd tests fail. But I havent seen this anywhere in pennylane so I am wondering if there is a cleaner way? 🤔

Reply:
If it says, sure! But I was hoping for a more elegant way to achieving this 😬 should be possible since for regular pennylane operators we can in fact do `np.array(0.5) * qml.PauliX(0)`, or is the problem that `PauliWord` / `PauliSentence` are inheriting from `dict`?

Reply:
Updated. Too much to leave the stackexchange link or good to be 100% transparent?

Reply:
Looks good to me.

Reply:
@Alex-Preciado anything to do here in terms of licensing? I guess not but better to check after todays retro